### ISSUE TITLE
KE Evaluation

### ISSUE THREAD

**Issue by Avmb** [THU-KEG/KEPLER#25]
--entity2id: a json file that maps entity names (in the dataset) to the ids in the entity embedding numpy file, where the key is the entity names in the dataset, and the value is the id in the numpy file.
--relation2id: a json file that maps relation names (in the dataset) to the ids in the relation embedding numpy file.

where do these files come from?

**Comment by DaliaDawod** [comment]
Did you figure it out? I am stuck at the same point

**Comment by DaliaDawod** [comment]
> --entity2id: a json file that maps entity names (in the dataset) to the ids in the entity embedding numpy file, where the key is the entity names in the dataset, and the value is the id in the numpy file. --relation2id: a json file that maps relation names (in the dataset) to the ids in the relation embedding numpy file.
> 
> where do these files come from?

Did you figure it out? I am stuck at the same point @Avmb

**Comment by brovatten** [comment]
> > --entity2id: a json file that maps entity names (in the dataset) to the ids in the entity embedding numpy file, where the key is the entity names in the dataset, and the value is the id in the numpy file. --relation2id: a json file that maps relation names (in the dataset) to the ids in the relation embedding numpy file.
> > where do these files come from?
> 
> Did you figure it out? I am stuck at the same point @Avmb

Did you figure it out? I am looking for the `relation2id.json` @DaliaDawod 🆘

**Comment by DaliaMedhat** [comment]
Yes I could after a looooong time, you can extract them from _**KEPLER/examples/KEPLER/Pretrain/convert.py**_ . You will find comments to lead you there @brovatten

**Comment by brovatten** [comment]
> KEPLER/examples/KEPLER/Pretrain/convert.p

My saviour, thank you so much Dalia! 🙏  You just saved me from a loooong time too :)

**Comment by Bakser** [comment]
Hi, sorry for the late reply. These two arguments are not for two fixed files. You are supposed to generate it for your used dataset as the comment introduced. All you need to do is to map the entity/relation names (like Q29387131/P31) to their indices (like 0,1) in the output embedding files generated by `examples/KEPLER/KE/generate_embeddings.py`.
